cocotb/clock.py

```diff
@@ -75,33 +75,43 @@ def __init__(self, signal, period, units=None):
         BaseClock.__init__(self, signal)
         self.period = get_sim_steps(period, units)
         self.half_period = get_sim_steps(period / 2.0, units)
-        self.frequency = 1.0 / get_time_from_sim_steps(self.period,units='us')
+        self.frequency = 1.0 / get_time_from_sim_steps(self.period, units='us')
         self.hdl = None
         self.signal = signal
         self.coro = None
         self.mcoro = None
 
     @cocotb.coroutine
-    def start(self, cycles=None):
+    def start(self, cycles=None, start_high=True):
         """Clocking coroutine.  Start driving your clock by forking a 
         call to this.
 
         Args:
             cycles (int, optional): Cycle the clock *cycles* number of times,
                 or if ``None`` then cycle the clock forever. 
                 Note: ``0`` is not the same as ``None``, as ``0`` will cycle no times.
+            start_high (bool, optional): Whether to start the clock with a ``1``
+                for the first half of the period.
+                Default is ``True``.
         """
         t = Timer(self.half_period)
         if cycles is None:
             it = itertools.count()
         else:
             it = range(cycles)
 
-        for _ in it:
-            self.signal <= 1
-            yield t
-            self.signal <= 0
-            yield t
+        if start_high:
+            for _ in it:
+                self.signal <= 1
+                yield t
+                self.signal <= 0
+                yield t
+        else:
+            for _ in it:
+                self.signal <= 0
+                yield t
+                self.signal <= 1
+                yield t
```

Comment:
Perhaps clearer, and definitely shorter, as:
```python
for _ in it:
    self.signal <= int(start_high)
    yield t
    self.signal <= int(not start_high)
    yield t
```

Reply:
I was wary that this might incur an additional runtime overhead, so played it safe.

Reply:
Possibly worth a comment like "branch outside for loop for performance"

Reply:
Good point, I have done so now.